### PR TITLE
Add plugin kill action

### PR DESCRIPTION
**Pull request by sp35**
# Description

- [x] Add `PluginActionViewSet` for extending to analyzers and connectors
- [x] Add `AnalyzerActionViewSet`
- [x] Add `ConnectorActionViewSet`
- [x] Update to store `task_id` in db instead of caching
- [x] Fix logic for job kill (mass `run_analyzer` kill)
- [x] Add unit tests

## Related issues
#534 (1. Plugin Kill)

# Checklist

- [x] The pull request is for the branch develop
- [x] I have added tests in the [Tests](https://github.com/intelowlproject/IntelOwl/blob/master/tests) folder. 
- [x] The tests gave 0 errors.
- [x] `Black` gave 0 errors.
- [x] `Flake` gave 0 errors.
- [ ] I squashed the commits into a single one. (optional, they will be squashed anyway by the maintainer)

# Real World Example
image
image

### CODE REVIEW

File: api_app/analyzers_manager/controller.py [hunk 1]
```diff
@@ -253,3 +240,19 @@ def run_analyzer(job_id: int, config: AnalyzerConfig, **kwargs) -> AnalyzerRepor
...
+    # kill celery tasks using task ids
+    task_ids = list(qs.values_list("task_id", flat=True))
+    celery_app.control.revoke(task_ids, terminate=True)
+
+    # update report statuses
+    statuses_to_filter = [
+        AnalyzerReport.Statuses.PENDING.name,
+        AnalyzerReport.Statuses.RUNNING.name,
+    ]
+    qs.filter(status__in=statuses_to_filter).update(
```
Comment (Eshaan7) [inline api_app/analyzers_manager/controller.py:256]: This `.filter` should be applied before calling `.revoke`. No point in revoking task IDs for analyzers that already finished.

File: api_app/connectors_manager/views.py [hunk 2]
```diff
@@ -3,11 +3,30 @@
 
 from rest_framework import generics
 from rest_framework.response import Response
+from django.http.response import Http404
```
Comment (Eshaan7) [inline api_app/connectors_manager/views.py:6]: `from rest_framework.exceptions import NotFound`

and then `raise NotFound()`, dont use `Http404`.

Comment (Eshaan7) [reply api_app/connectors_manager/views.py:6]: same in `analyzers_manager.views`.

File: api_app/analyzers_manager/controller.py [hunk 3]
```diff
@@ -27,10 +26,6 @@
 DEFAULT_QUEUE = "default"
 
 
-def build_cache_key(job_id: int) -> str:
-    return f"job.{job_id}.analyzers_manager.task_ids"
-
-
 def filter_analyzers(serialized_data: Dict, warnings: List) -> List[str]:
```
Comment (Eshaan7) [inline api_app/analyzers_manager/controller.py:29]: `-> Dict[str, str]` after new changes.

Comment (sp35) [reply api_app/analyzers_manager/controller.py:29]: I guess you meant
```python
def start_analyzers(
    job_id: int,
    analyzers_to_execute: List[str],
    runtime_configuration: Dict[str, Dict] = None,
) -> Dict[str, str]:
```

Comment (Eshaan7) [reply api_app/analyzers_manager/controller.py:29]: Right! sorry.

**Review by Eshaan7 — APPROVED** [review]

**Pull request merged**